src/gzip/dynamic_compression_handler.cr

```diff
@@ -0,0 +1,57 @@
+{% if !flag?(:without_zlib) %}
+  require "gzip"
+{% end %}
+
+module Lucky::DynamicCompressionHandler
+  COMPRESS_ONLY_THESE_EXTENSIONS = ["", ".html", ".htm", ".txt", ".json", ".xml"]
```

Comment:
Why is there a blank one at the beginning? Is that like for paths with no extension (default).

Reply:
Yep.

Reply:
Actually I just realized that  maybe other than compressing using an extension it might be more reliable to use the `Content-Type` header. Thoughts?

Reply:
I don't think Content-Type has been set yet at this point. Basically we are wrapping the IO with the Gzip::Writer before the render happens. Let me do a little testing to make sure I'm right about that.

Reply:
Yeah, I was right about that. We only have headers available from the request at this point.

Reply:
Ah I see. Ok I have a proposal if you are up for it. We could do 2 PRs. Once for static file handler and one for the dynamic one. My hunch is the static one will be simpler and we can get that in pretty quick. The dynamic one may be a bit trickier to really nail down but if it is a separate PR it won't block the static one.

Thoughts?

Reply:
Another idea is that we set the GZIP writer in renderable instead of in a handler. https://github.com/luckyframework/lucky/blob/master/src/lucky/text_response.cr

Reply:
I agree with 2 PRs. I had thought the same thing, but I just did the one PR here to give an idea of the combined strategy.

I'll look into setting the writer in that location before doing the PR for the dynamic handler.

Reply:
Oh perfect! Sounds like you've already got this done. Stoked to see the PRs. Thanks for thoroughly researching all the options. I like this a lot more than where we were previously going with doing gzip at runtime 👍. Also cool that people can use other compression like Brotli if they want